### DISCUSSION TITLE
It's a unknown error when I compile any example of STM32duino_Low_Power with LTO

### DISCUSSION THREAD

**Discussion by sunnyguhz** [stm32duino/Arduino_Core_STM32#2451] [Q&A] [answered]
**Describe the bug**
When I compile any example of STM32duino_Low_Power with LTO, it's has a unknown error to make it stop.
And it's no problem without LTO.

**To Reproduce**
```
#include "STM32LowPower.h"

void setup() {
  pinMode(LED_BUILTIN, OUTPUT);
  // Configure low power
  LowPower.begin();
}

void loop() {
  digitalWrite(LED_BUILTIN, HIGH);
  LowPower.deepSleep(1000);
  digitalWrite(LED_BUILTIN, LOW);
  LowPower.deepSleep(1000);
}
```

**Screenshots**
lto1.exe: fatal error: open C:\Users\Sunnygu\AppData\Local\Temp\arduino\sketches\6AF94743C3DEA696995DF90E3A49899D\core\generic_clockĬ��.c.o failed: Unknown error
compilation terminated.
lto-wrapper.exe: fatal error: C:\Users\Sunnygu\AppData\Local\Arduino15\packages\STMicroelectronics\tools\xpack-arm-none-eabi-gcc\13.2.1-1.1/bin/arm-none-eabi-gcc returned 1 exit status
compilation terminated.
C:/Users/Sunnygu/AppData/Local/Arduino15/packages/STMicroelectronics/tools/xpack-arm-none-eabi-gcc/13.2.1-1.1/bin/../lib/gcc/arm-none-eabi/13.2.1/../../../../arm-none-eabi/bin/ld.exe: error: lto-wrapper failed
collect2.exe: error: ld returned 1 exit status

Using library STM32duino Low Power at version 1.2.5 in folder: C:\Users\Sunnygu\Documents\Arduino\libraries\STM32duino_Low_Power 
Using library STM32duino RTC at version 1.4.0 in folder: C:\Users\Sunnygu\Documents\Arduino\libraries\STM32duino_RTC 
Using library SrcWrapper at version 1.0.1 in folder: C:\Users\Sunnygu\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.8.0\libraries\SrcWrapper 
exit status 1

Compilation error: exit status 1

**Desktop (please complete the following information):**
OS: [Windows10 22H2]
Arduino IDE version: [2.3.2]
STM32 core version: [2.8.0]
Tools menu settings if not the default: [No Serial]
Upload method: [SWD]

**Board (please complete the following information):**
 FQBN: STMicroelectronics:stm32:GenG0:pnum=GENERIC_G030C8TX,xserial=disabled

**Comment by fpistm** [comment]
LTO is provided as is. It is somehow experimental. I will check when got some free time.

**Reply by sunnyguhz** [reply]
Thank you so much.

**Reply by fpistm** [reply, marked as answer]
Hi @sunnyguhz 
I've tested to build your sketch for the same board with LTO enabled and I didn't have any error.

```console
Using library STM32duino Low Power at version 1.2.5 in folder: C:\IDE\data\Arduino\libraries\STM32duino_Low_Power 
Using library STM32duino RTC at version 1.4.0 in folder: C:\IDE\data\Arduino\libraries\STM32duino_RTC 
Using library SrcWrapper at version 1.0.1 in folder: C:\Users\***\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.8.1\libraries\SrcWrapper 
"C:\\Users\\***\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\tools\\xpack-arm-none-eabi-gcc\\13.2.1-1.1/bin/arm-none-eabi-size" -A "C:\\Users\\***\\AppData\\Local\\Temp\\arduino\\sketches\\E6E750533C7C25631890A7DE3ECC27E4/sketch_jul19a.ino.elf"
Sketch uses 9108 bytes (13%) of program storage space. Maximum is 65536 bytes.
Global variables use 936 bytes (11%) of dynamic memory, leaving 7256 bytes for local variables. Maximum is 8192 bytes.
```

The only difference is the OS. I'm on Windows 11 now. Maybe linked to xpack arm none eabi binaries but in that case can't do nothing it have to be reported to the package maintainer.
